source/develop/release-management/features/virt/enhance-import-export-with-ova.md

```diff
@@ -0,0 +1,135 @@
+
+## An Enhancement for Import OVAs
+This feature enhances the existing support for importing OVA files that are accessible to the host, by enabling to specify a path to an OVA file that was generated by oVirt. Importing an OVAfile that was generated by oVirt does not involve the adaptation needed for OVA files generated by other platforms (e.g., installing kvm-specific drivers) and thus, should be faster than importing of such OVA files.  
+
+## Downloading an OVA
+This feature will leverage the implementation of the planned download-image to provide similar functionality for OVA files (i.e., download a VM/template as OVA on the client's machine).  
+
+# Design
+The following section describes the high-level design for the different aspects described in the previous section.
+
+## Uploading an OVA
+In principle, uploading an OVA is similar to uploading an image. However, it is more challenging because OVA is a more complex resource than an image.  
+
+The upload process would be implemented differently depending on whether the OVA was generated by oVirt (or more accurately, that the disks within the OVA are of type raw or qcow) or by other source (and thus requires adaptation). In both cases, the user will be able to select a file and then be provided with the VM configuration that allows him to configure the parameters for the import process.  
+
+### Uploading an OVA Generated by oVirt
+The following figure depicts the import process of an OVA that was generated by oVirt:
+
+![](../../../../images/wiki/upload-ova.png)
+
+1. The front-end retrieves the OVF configuration from the OVA.
+2. The front-end sends the OVF configuration to the back-end.
+3. The back-end parses the OVF configuration and returns a VM to the front-end.
+4. The front-end sends (an optionally customized) parameters of the import process to the back-end.
+5. The back-end persists the VM.
+6. The back-end triggers the creation of the target VM disks on the storage domain.
+7. The back-end sends the import parameters, including the destination disks, to imageio-proxy, that sends them to imageio.
+8. Imageio-proxy sends the engine a URL to write to. This URL is sent to the front-end.
+9. The front-end streams the **whole OVA** to imageio-proxy, that streams it to imageio.
+10. Imageio reads the data and according to the import parameters streams the data into the corresponding target disks.
+
+Notes:
+
+1. The described process assumes that the disks are not compressed within the OVA. In order to support this, the back-end would need to pass an indication to imageio-proxy of whether or not each disk is compressed (step 7) so the proxy could uncompress the data on-the-fly while streaming it to the destination volume (step 10).
+2. In the figure above the format of the disks within the OVA is QCOW that is going to be the default format in OVA files generated by oVirt. However, we should support import of OVA that contains RAW disks as well.
```

Comment:
Suggestion:
In the figure above, the format of the disks in the OVA file is QCOW. This will serve as the default format in all OVA files generated by oVirt. However, we should also support the import of OVA files containing RAW format disks.